### PR TITLE
CAPETOWN - CLASS_1 - SIMBA PFAIRA - JAVASCRIPT CORE 1 - WEEK 2

### PR DESCRIPTION
**Pull request by sirpfaira**
Capetown Class 1 - Simba Pfaira - Javascript Core 1 - Week 2

-->

# Your Details

- Your Name: Simba Pfaira
- Your City: Capetown SA
- Your Slack Name: Simba Pfaira

# Homework Details

- Module: Javascript Core 1
- Week: 2

### CODE REVIEW

**Comment by bradleymubenga777** [comment]
Hey Simba! I noticed that you did not include your homework details in the pull request. Doing so would be very helpful to our Teachers, thanks!

**Comment by RodrigueNdzana** [comment]
Hi Simba, well done o completing the exercise. The syntax of your if statement are not in correct format  according on how to write an if statement in javaScript. Have a look at this link on how anif statement is structure in JavaScript.

https://www.w3schools.com/js/js_if_else.asp#:~:text=In%20JavaScript%20we%20have%20the,the%20first%20condition%20is%20false

**Comment by sirpfaira** [comment]
> Hey Simba! I noticed that you did not include your homework details in the pull request. Doing so would be very helpful to our Teachers, thanks!

ok noted thank you

File: mandatory/2-function-creation.js [hunk 1]
```diff
@@ -5,7 +5,15 @@ Write a function that:
 - removes any forward slashes (/) in the strings
 - makes the string all lowercase
 */
-function tidyUpString(strArr) {}
+function tidyUpString(strArr) {
```
Comment (Ntshembo-Hlongwane1) [inline mandatory/2-function-creation.js:8]: ### Hey there so I just have some code refactoring tips for you.

- So I am pretty sure your **tidyUpString function** does pass all test cases, however we can refactor it a bit
- So good thing you are already looping and indexing through each each element
- So I do not know if you know that in JS we can chain methods like this below

```js
.replace("/", "").trim().toLowerCase();
```

```diff
- let trimmedStr = strArr[x].trim();
- let tidyStr = trimmedStr.replace("/", '') ;
- let lowerCase = tidyStr.toLowerCase(); 
+ strArr[x] = strArr[x].replace("/", "").trim().toLowerCase();
```
- Now all those 3 lines of code are just boiled down to just **1 single line**

### However we could still work on this code a bit more by making our lives easier on the looping side 

```js
let count = 0;
for (string of strArr){
    strArr[count] = string.replace("/", "").trim().toLowerCase();;
    count++;
};
```
[Refer to this link to check how that kind of for loop work's it is a game changer in certain cases](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of)

### Hope this was helpful 🔥💪

**Pull request closed without merging**